### ISSUE TITLE
When using dom.watch() icon, sizing and fa-fw does not apply

### ISSUE THREAD

**Issue by rostamiani** [FortAwesome/vue-fontawesome#226]
When using *dom.watch()* icon, sizing and *fa-fw* does not apply. 

This is my main.js:
``` javascript
import { dom } from "@fortawesome/fontawesome-svg-core";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faCompress, faCompressAlt, faExpandAlt, faArrowsAlt, faPencilAlt } from "@fortawesome/free-solid-svg-icons";

// Load fontawesome icons
library.add(faPencilAlt, faCompress, faArrowsAlt, faCompressAlt, faExpandAlt);
dom.watch();
```

And this is my template:
``` html
<i class="fas fa-compress fa-lg fa-fw"></i>
```
And the output is still small

**Comment by robmadole** [comment]
@rostamiani using `dom.watch()` is not the recommended way of using Font Awesome with a Vue project. Can you tell me a little bit about why you are doing this?

**Comment by rostamiani** [comment]
@robmadole Because I have a VS Code extension that inserts 'i' tags automatically and I don't have to memorize all class names. This really helps.
But in the normal method I couldn't find any extension to do this for me.

Is there any performance problem using _dom.watch()_?

**Comment by robmadole** [comment]
@rostamiani yes, `dom.watch()` is not compatible with Vue. It mutates the DOM at the same time Vue is trying to control it. Causes lots of nasty problems and it's why we created this component.

If you'd like to stick with `<i>` tags I would suggest you remove `vue-fontawesome` and switch to a web fonts version of Font Awesome. You can install `@fortawesome/fontawesome-free` or `@fortawesome/fontawesome-pro` (if you are a subscriber) and inside those packages are `css/*.css` files that point to `webfont/*.*` files. You should be able to import the `css` files and if you have modern tooling (Webpack) it should know what to do with those.

If using an `import` gives you problems you can go to a simpler integration using fontawesome.com/kits or host the files from your static assets directory by downloading them from fontawesome.com/download.

**Comment by pholly** [comment]
@robmadole  I'm new to vue-fontawesome. I'm confused because [Readme](https://github.com/FortAwesome/vue-fontawesome#processing-i-tags-into-svg-using-font-awesome) states we need to use `dom.watch`.  Maybe that was copied from npm instructions by accident?

The font awesome vue components should take care of rendering svg right? So we shouldn't use `dom.watch`?

**Comment by mgmedick** [comment]
Just adding on to what @pholly said, the official "vue-fontawesome" documentation [](https://github.com/FortAwesome/vue-fontawesome) mentions ability to use dom.watch(), and says nothing about it being detrimental to performance. If the official stance is not to use dom.watch() with vue can you update that documentation please.

Also  just want some clarification, does this mean that font-awesome svg core is unusable for someone using Vue that needs to keep the <i> elements? In my case where the vue-fontawesome approach fails for me me is returning an html string in a tabulator table, which is then embedded into the table. So with the <i> tag and dom.watch, the replace works, but with <font-awesome-icon>  (and no dom.watch) it doesn't. I can't think of a way of getting it to work vue-fontawesome with  since i'm not adding the actual element to the dom, I'm returning html string which is then embedded in table.

**Comment by jasonlundien** [comment]
Closing older issue.
